src/v/kafka/server/group.cc

```diff
@@ -2296,4 +2328,28 @@ ss::sstring group_state_to_kafka_name(group_state gs) {
     }
 }
 
+void group::add_pending_member(
+  const kafka::member_id& member_id, duration_type timeout) {
+    auto res = _pending_members.try_emplace(member_id, [this, member_id] {
```

Comment:
can franz-go/internals allow you to manipulate this state for a ducktape test harness for this condition.

Reply:
ah i see the boost test. nvm.